lib/rules/media-feature-name-allowed-list/index.js

```diff
@@ -35,40 +33,28 @@ const rule = (primary) => {
 		}
 
 		root.walkAtRules(/^media$/i, (atRule) => {
-			mediaParser(atRule.params).walk(/^media-feature$/i, (mediaFeatureNode) => {
-				const parent = mediaFeatureNode.parent;
-				const mediaFeatureRangeContext = isRangeContextMediaFeature(parent.value);
+			const hasComments = atRule.raws.params && atRule.raws.params.raw;
+			const mediaRule = hasComments ? hasComments : atRule.params;
```

Comment:
[suggest] The same suggestion as <https://github.com/stylelint/stylelint/pull/6850#discussion_r1198839060>:

```suggestion
			const mediaRule = getAtRuleParams(atRule);
```

Reply:
Done :)